### PR TITLE
Update drop backend behavior

### LINKED ISSUE #1576: Update drop backends syntax behavior
Update drop backends syntax behavior.
Added syntax `ALTER SYSTEM DROP BACKEND <backends> FORCE`.
## Feature request

**Is your feature request related to a problem? Please describe.**

When a tablet with a single duplication table exists on a backend, deleting backend will cause table data loss.

**Describe the solution you'd like**

I think the drop backends operation is rarely used in online clusters. And ‘dropp’ is not a correct word.
Make the following changes to the behavior of the drop backend statement:
- `ALTER SYSTEM DROP <backends>` can delete backend instead of `DROPP`.
- When there is a tablet with a single duplication table on the backend to be deleted, an exception will be thrown and `ALTER SYSTEM DROP <backends> FORCE` needs to be used to delete.
- The original behavior of `DROPP` remains unchanged, I think it could be removed in a future version.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by stephen-shelby**
`DROP BACKEND` will drop all tablets on the backend. If some tables have only one replica, the data will be lost.
The pull request is to add a prompt when necessary. As well, `DROP BACKNE <backend> FORCE` can be used to forcibly drop the backend.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 1]
```diff
@@ -191,16 +198,50 @@ public void dropBackend(long backendId) throws DdlException {
...
+                .filter(table -> table.getType() == Table.TableType.OLAP)
+                .map(table -> (OlapTable) table)
+                .filter(table -> table.getTableProperty().getReplicationNum() == 1)
+                .forEach(table -> {
+                    table.getAllPartitions().forEach(partition -> {
+                        boolean existIntersection = partition.getBaseIndex().getTablets().stream()
+                                .map(Tablet::getId).anyMatch(tabletIds::contains);
+
+                        if (existIntersection) {
+                            String errMsg = String.format("There is a single duplication of the olap table " +
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java]: done

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -1034,6 +1034,10 @@ alter_system_clause ::=
         RESULT = new DropBackendClause(hostPorts, false);
     :}
     | KW_DROPP KW_BACKEND string_list:hostPorts
+    {:
+        RESULT = new DropBackendClause(hostPorts, true, true);
+    :}
+    | KW_DROP KW_BACKEND string_list:hostPorts KW_FORCE
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/cup/sql_parser.cup]: @chaoyli Has been compatible with drop partition/schema/table/database.

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java]: @chaoyli done

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 3]
```diff
@@ -191,16 +198,52 @@ public void dropBackend(long backendId) throws DdlException {
...
-        dropBackend(backend.getHost(), backend.getHeartbeatPort());
+        dropBackend(backend.getHost(), backend.getHeartbeatPort(), false);
+    }
+
+    private void checkUnforce(Backend droppedBackend) {
+        Catalog catalog = Catalog.getCurrentCatalog();
+        List<Long> tabletIds = Catalog.getCurrentInvertedIndex().getTabletIdsByBackendId(droppedBackend.getId());
+        List<Long> dbs = catalog.getDbIds();
+
+        dbs.stream().map(catalog::getDb).forEach(db ->
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:209]: add db read lock

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java]: @gengjun-git done, I will remind it later.

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**